app-modules/ai/src/Filament/Resources/PromptResource/Pages/EditPrompt.php

```diff
@@ -87,6 +91,32 @@ public function form(Form $form): Form
                             ])
                             ->grouped()
                             ->visible(auth()->user()->hasRole(Authenticatable::SUPER_ADMIN_ROLE)),
+                        Group::make()->schema([
+                            Checkbox::make('is_confidential')
+                                ->label('Confidential')
+                                ->live()
+                                ->columnSpanFull(),
+                            Select::make('confidential_prompt_users')
+                                ->relationship('confidentialAccessUsers', 'name')
+                                ->preload()
+                                ->label('Users')
+                                ->multiple()
+                                ->exists('users', 'id')
+                                ->visible(fn (Get $get) => $get('is_confidential')),
+                            Select::make('confidential_prompt_teams')
+                                ->relationship('confidentialAccessTeams', 'name')
+                                ->preload()
+                                ->label('Teams')
+                                ->multiple()
+                                ->exists('teams', 'id')
+                                ->visible(fn (Get $get) => $get('is_confidential')),
+                        ])
+                            ->columns(2)
+                            ->columnSpanFull()
+                            ->visible(fn (Get $get) => ConfidentialPromptsFeature::active()
+                                && auth()->user()->hasRole(Authenticatable::SUPER_ADMIN_ROLE)
+                                || auth()->user()->id == $get('user_id')
```

Comment:
I am not sure this `$get('user_id')` call works? `$get` only retrieves values from this form, not from the record. You will probably need to load in the record and check with that.

You should have tests for this functionality and access, which would have caught this.